### PR TITLE
Added Data Sources

### PR DESCRIPTION
**Pull request by airon-applyinnovations**
Related to #22

Added data sources

- `neon_project` - fetches neon project details given project id 
- `neon_branches` - fetches branches given project id
- `neon_branch_endpoints` - fetches endpoints (DB host) given project id and branch id
- `neon_branch_roles` - fetches available users/roles given project id and branch id
- `neon_branch_role_password` - fetches the password given project id, branch id and role name

I also updated acceptance test to test the new data sources and run `make docu` to update docs. 

There are some formatting changes included in this pull request due to `go fmt` removing some white spaces.

### CODE REVIEW

**Comment by alexander-applyinnovations** [comment]
@kislerdm we are currently using this code for a production project and would appreciate if you could review, happy to incorporate any feedback

File: internal/provider/data_source_branches.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func dataSourceBranches() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Fetch Project Branches.",
+		SchemaVersion: 8,
```
Comment (kislerdm) [inline internal/provider/data_source_branches.go:16]: **(issue)** AFAIK schema version is referring to the schema of the resource it defines. Unfortunately, initially I used it wrong in the beginning 😄 

```suggestion
                SchemaVersion: 1,
```

File: internal/provider/data_source_project.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func dataSourceProject() *schema.Resource {
+	return &schema.Resource{
+		Description:   `Fetch Project.`,
+		SchemaVersion: 8,
```
Comment (kislerdm) [inline internal/provider/data_source_project.go:15]: **(issue)** AFAIK schema version is referring to the schema of the resource it defines. Unfortunately, initially I used it wrong in the beginning 😄 

```suggestion
                SchemaVersion: 1,
```

**Review by kislerdm — APPROVED** [review]
@airon-applyinnovations @alexander-applyinnovations Hello Airon and Alexander! Thank you very much for your contribution! I'll include it in the next release.

~~I left a tiny nitpick/issue - please apply suggestions. Thanks!~~

File: internal/provider/data_source_branch_role_password.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func dataSourceBranchRolePassword() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Fetch Role Password.",
+		SchemaVersion: 8,
```
[the remark this answers is not in the document]
Comment (kislerdm) [reply internal/provider/data_source_branch_role_password.go:16]: I'll followup on it after merge to trunk

File: internal/provider/data_source_branch_roles.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func dataSourceBranchRoles() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Fetch Branch Roles.",
+		SchemaVersion: 8,
```
[the remark this answers is not in the document]
Comment (kislerdm) [reply internal/provider/data_source_branch_roles.go:16]: I'll followup on it after merge to trunk

File: internal/provider/data_source_project.go (hunk 2 above)
Comment (kislerdm) [reply internal/provider/data_source_project.go:15]: I'll followup on it after merge to trunk

File: internal/provider/data_source_branch_endpoints.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	neon "github.com/kislerdm/neon-sdk-go"
+)
+
+func dataSourceBranchEndpoints() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Fetch Branch Endpoints",
+		SchemaVersion: 8,
```
[the remark this answers is not in the document]
Comment (kislerdm) [reply internal/provider/data_source_branch_endpoints.go:15]: I'll followup on it after merge to trunk

File: internal/provider/data_source_branches.go (hunk 1 above)
Comment (kislerdm) [reply internal/provider/data_source_branches.go:16]: I'll followup on it after merge to trunk

**Pull request merged**